lib/galaxy/datatypes/sniff.py

```diff
@@ -780,9 +780,7 @@ def handle_uploaded_dataset_file_internal(
                 ext = guess_ext(converted_path, sniff_order=datatypes_registry.sniff_order, is_binary=is_binary)
         else:
             ext = guessed_ext
-
-        if not is_binary and check_content and check_html(converted_path):
-            raise InappropriateDatasetContentError('The uploaded file contains invalid HTML content')
+            
```

Comment:
This is pretty intentional - allowing HTML uploads introduces security problems we don't want I think. There is a bunch of things we could potentially do like serve HTML content on subdomains and such to make this more secure but I don't think we're there yet in terms of the infrastructure.